File: Scripts/rpi/function-composition/c++/function-composition.cpp
#include <iostream>
#include <cmath>
#include <ext/functional>

int main() {
  for (int i = 0; i < 100000000; ++i) {
     __gnu_cxx::compose1(std::ptr_fun(::sin), std::ptr_fun(::asin))(0.5);
  }
  return 0;
}

File: Scripts/rpi/function-composition/perl/function-composition.pl
sub compose {
    my ($f, $g) = @_;

    sub {
        $f -> ($g -> (@_))
    };
}

use Math::Trig;
for($a = 0; $a < 100000000; $a += 1) { 
    compose(sub {sin $_[0]}, \&asin)->(0.5)
}
